style-guide.md

```diff
@@ -0,0 +1,38 @@
+## Style Guide
+
+Contributions to this repository should follow the style guide as described in this section.
+
+### Markdown
+
+Markdown files in this project should follow the style enforced by the [markdownlint tool](https://www.npmjs.com/package/markdownlint),
+as configured by the `.markdownlint.yaml` file in the root of the project.
+The `markdownlint` tool can also fix formatting, which can save time with tables in particular.
+
+The following additional rules should be followed but currently are not enforced by tooling:
+
+1. The first mention of a normative reference or an OAS-defined Object in a (sub)*section is a link, additional mentions are not.
+2. OAS-defined Objects such as Schema Objects are written in this style, and are not monospaced.
+3. Use "example" instead of "sample" - this spec is not about statistics.
+4. Use "OpenAPI Object" instead of "root".
+5. Fixed fields are monospaced.
+6. Field values are monospaced in JSON notation: `true`, `false`, `null`, `"header"` (with double-quotes around string values).
+7. A combination of fixed field name with example value uses JS notation: `in: "header"`, combining rules 5 and 6.
+8. An exception to 5-7 is colloquial use, for example "values of type `array` or `object`" - "type" is not monospaced, so the monospaced values aren't enclosed in double quotes.
```

Comment:
Can we add some note that "primitive" is not a formal type and should not be written `primitive`?  We messed this up in the patch releases in a few places.

Reply:
I'm not sure how to provide an exhaustive list of words that shouldn't be formatted. The list above should help us just format the ones we do want, I think?

Reply:
Oh, I definitely wouldn't try to be exhaustive, and maybe there's a better way to handle it than a word list.

The [problem we have in the new patch releases](https://spec.openapis.org/oas/v3.1.1#style-values) is that `primitive` looks like a literal `type` value impacting the `style` field, when really it's a category encompassing the literal values `number`, `integer`, `string`, `boolean`, and `"null"`.  That's confusing, although we do use "primitives" with an [example using a `string`](https://spec.openapis.org/oas/v3.1.1#example-basic-multipart-form) and in [an example in another section](https://spec.openapis.org/oas/v3.1.1#primitive-example).

Ideally, our style guide would help us avoid these category-vs-literal-value errors (although I don't know that we have any other such errors).